Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1373 @@
+
+                UnlockStream();
+                return false;
+            }
+            
+            
+            if ((string)orderResponse["status"] == "success")
+            {
+                if (string.IsNullOrEmpty((string)orderResponse["data"]["order_id"]))
+                {
+                    var errorMessage = $"Error parsing response from modify order: {orderResponse["status_message"]}";
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Zerodha Update Order Event") { Status = OrderStatus.Invalid, Message = errorMessage });
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning,(string) orderResponse["status"], errorMessage));
+
+                    UnlockStream();
+                    return true;
+                }
+
+                var brokerId = (string)orderResponse["data"]["order_id"];
+                if (CachedOrderIDs.ContainsKey(order.Id))
+                {
+                    CachedOrderIDs[order.Id].BrokerId.Clear();
+                    CachedOrderIDs[order.Id].BrokerId.Add(brokerId);
+                }
+                else
+                {
+                    order.BrokerId.Add(brokerId);
+                    CachedOrderIDs.TryAdd(order.Id, order);
+                }
+
+                // Generate submitted event
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Zerodha Update Order Event") { Status = OrderStatus.UpdateSubmitted });
+                Log.Trace($"Order modified successfully - OrderId: {order.Id}");
+
+                UnlockStream();
+                return true;
+            }
+
+            var message = $"Order failed, Order Id: {order.Id} timestamp: {order.Time} quantity: {order.Quantity} content: {orderResponse["status_message"]}";
+            OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Zerodha Update Order Event") { Status = OrderStatus.Invalid });
+            OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, -1, message));
+
+            UnlockStream();
+            return false;
+        }
+
+        /// <summary>
+        /// Cancels the order with the specified ID
+        /// </summary>
+        /// <param name="order">The order to cancel</param>
+        /// <returns>True if the request was submitted for cancellation, false otherwise</returns>
+        public override bool CancelOrder(Order order)
+        {
+            LockStream();
+            Log.Trace("CancelOrder Order Status "+order.Status.ToString());
+            //if (order.Type == OrderType.Bracket)
+            //{
+            //    orderResponse = _kite.CancelOrder(order.Id.ToStringInvariant());
+            //}
+            //else
+            //{
+            JObject orderResponse = new JObject();
+            if (order.Status!=OrderStatus.Filled && order.Status!= OrderStatus.PartiallyFilled && order.Status!=OrderStatus.Invalid)
```

Comment:
I think partially filled orders can still be cancelled for the unfilled quantity?

Maybe this could be better:

``` C#
if (order.Status.IsOpen()) { ... }
```